test/functional/lua/xdiff_spec.lua

```diff
@@ -0,0 +1,103 @@
+local helpers = require('test.functional.helpers')(after_each)
+local clear = helpers.clear
+local exec_lua = helpers.exec_lua
+local eq = helpers.eq
+local pcall_err = helpers.pcall_err
+
+describe('xdiff bindings', function()
+  before_each(function()
+    clear()
+  end)
+
+  describe('can diff text', function()
+    before_each(function()
+      exec_lua[[
+        a1 = 'Hello\n'
+        b1 = 'Helli\n'
+
+        a2 = 'Hello\nbye\nfoo\n'
+        b2 = 'Helli\nbye\nbar\nbaz\n'
+      ]]
+    end)
+
+    it('with no callback', function()
+
+      eq(
+        table.concat({
+          '@@ -1 +1 @@',
+          '-Hello',
+          '+Helli',
+          ''
+        }, '\n'),
+        exec_lua("return vim.xdl_diff(a1, b1)")
+      )
+
+      eq(
+        table.concat({
+          '@@ -1 +1 @@',
+          '-Hello',
+          '+Helli',
+          '@@ -3 +3,2 @@',
+          '-foo',
+          '+bar',
+          '+baz',
+          ''
+        }, '\n'),
+        exec_lua("return vim.xdl_diff(a2, b2)")
+      )
+
+    end)
+
+    it('with callback', function()
+      exec_lua([[hunk_func = function(sa, ca, sb, cb)
+          exp[#exp+1] = {sa, ca, sb, cb}
+        end]])
+
+      exec_lua("exp = {}")
+      exec_lua([[return vim.xdl_diff(a1, b1, {hunk_func = hunk_func})]])
+      eq({{0, 1, 0, 1}}, exec_lua("return exp"))
+
+      exec_lua("exp = {}")
+      exec_lua([[return vim.xdl_diff(a2, b2, {hunk_func = hunk_func})]])
+      eq({{0, 1, 0, 1}, {2, 1, 2, 2}}, exec_lua("return exp"))
```

Comment:
even here the `exec_lua` should be merged. exec_lua takes a _block_ of lua code, not a single statement.

Reply:
Done